UM/Scene/SceneNode.py

```diff
@@ -590,72 +561,38 @@ def setCalculateBoundingBox(self, calculate):
-        self._derived_scale = None
-
         self.transformationChanged.emit(self)
 
         for child in self._children:
             child._transformChanged()
 
     def _updateTransformation(self):
-        scale_and_mirror = self._scale * self._mirror
-        self._transformation = Matrix.fromPositionOrientationScale(self._position, self._orientation, scale_and_mirror)
-
+        scale, shear, euler_angles, translation = self._transformation.decompose()
+        self._position = translation
+        self._scale = scale
+        self._shear = shear
+        orientation = Quaternion()
+        euler_angle_matrix = Matrix()
+        euler_angle_matrix.setByEuler(euler_angles.x, euler_angles.y, euler_angles.z)
+        orientation.setByMatrix(euler_angle_matrix)
+        self._orientation = orientation
         if self._parent:
-            parent_orientation = self._parent._getDerivedOrientation()
-            if self._inherit_orientation:
-                self._derived_orientation = parent_orientation * self._orientation
-            else:
-                self._derived_orientation = self._orientation
+            self._world_transformation = self._parent.getWorldTransformation().multiply(self._transformation, copy = True)
+        else:
+            self._world_transformation = self._transformation
 
-            # Sometimes the derived orientation can be None.
-            # I've not been able to locate the cause of this, but this prevents it being an issue.
-            if not self._derived_orientation:
-                self._derived_orientation = Quaternion()
+        world_scale, world_shear, world_euler_angles, world_translation = self._world_transformation.decompose()
+        self._derived_position = world_translation
+        self._derived_scale = world_scale
 
-            parent_scale = self._parent._getDerivedScale()
-            if self._inherit_scale:
-                self._derived_scale = parent_scale.scale(scale_and_mirror)
-            else:
-                self._derived_scale = scale_and_mirror
+        world_euler_angle_matrix = Matrix()
+        world_euler_angle_matrix.setByEuler(world_euler_angles.x, world_euler_angles.y, world_euler_angles.z)
+        self._derived_orientation.setByMatrix(world_euler_angle_matrix)
 
-            self._derived_position = parent_orientation.rotate(parent_scale.scale(self._position))
-            self._derived_position += self._parent._getDerivedPosition()
-
-            self._world_transformation = Matrix.fromPositionOrientationScale(self._derived_position, self._derived_orientation, self._derived_scale)
-        else:
-            self._derived_position = self._position
-            self._derived_orientation = self._orientation
-            self._derived_scale = scale_and_mirror
-            self._world_transformation = self._transformation
+        world_scale, world_shear, world_euler_angles, world_translation = self._world_transformation.decompose()
```

Comment:
Here you decompose the world transformation again but don't do anything with the results, and `self._world_transformation` hasn't changed since the previous time. I think this might have been a merge conflict?